src/components/Messaging/MessageBody.js

```diff
@@ -333,22 +364,23 @@ const MessageBody = ({ messages, messageRecipient }) => {
   const { paper, messagesBody, textInput } = useStyles();
   const { user, authors, postMessageSend } = useContext(MessagingContext);
   const [messageContent, setMessageContent] = useState('');
-  const [subject, setSubject] = useState('');
+  const [subject, setSubject] = useState(messages ? messages[0].subject : '');
   const [recipientId, setRecipientId] = useState('');
 
   useEffect(() => {
     if (messages) {
+      // messages are either Surveys or Messages/Announce Messages
       setSubject(messages[0].subject);
     }
-  }, [messages]);
+  }, [messages, messageRecipient]);
 
   useEffect(() => {
-    if (authors && messageRecipient) {
-      let res = authors.find((author) => author.handle === messageRecipient);
-
-      if (res) {
-        setRecipientId(res.id);
-      }
+    const res = authors.find((author) => author.handle === messageRecipient);
+    if (res?.id) {
+      setRecipientId(res.id);
+    } else {
+      // if not an author it must be an org or a region and the id is the handle
+      setRecipientId(messageRecipient);
     }
```

Comment:
If the recipient is an org or a region, there is no recipient id shows to the sender of that message.  Rather this is populated by the bulk_message_recipients property, which an an array describing the recipient parameters.

Reply:
Wouldn't it be useful for the sender to see who they sent that message to, if they have more than one option to target? which region? which org?